src/commons/withScrollEnabler.tsx

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import React, {useState, useEffect, useCallback, useRef} from 'react';
+// eslint-disable-next-line no-unused-vars
+import {FlatListProps, ScrollViewProps} from 'react-native';
+import {Constants} from 'react-native-ui-lib';
+import forwardRef from './forwardRef';
+
+export type WithScrollEnablerProps = FlatListProps<any> | ScrollViewProps;
+function withScrollEnabler<PROPS extends WithScrollEnablerProps>(WrappedComponent: React.ComponentType<WithScrollEnablerProps>): React.ComponentType<PROPS> {
+  const ScrollEnabler = (props: WithScrollEnablerProps) => {
+    const [scrollEnabled, setScrollEnabled] = useState(true);
+    const [contentSize, setContentSize] = useState(0);
+    const [layoutSize, setLayoutSize] = useState(0);
+
+    const scrollViewRef = useRef<WithScrollEnablerProps>(null);
+
+    useEffect(() => {
+      const {forwardedRef} = props;
+      if (_.isFunction(forwardedRef)) {
+        forwardedRef(scrollViewRef);
+      } else if (_.isObject(forwardedRef)) {
+        forwardedRef.current = scrollViewRef;
+      }
+    }, [scrollViewRef]);
+
+    useEffect(() => {
+      const isScrollEnabled = contentSize > layoutSize;
+      if (isScrollEnabled !== scrollEnabled) {
+        setScrollEnabled(isScrollEnabled);
+        if (Constants.isAndroid && scrollViewRef.current && scrollViewRef.current) {
+          if (_.isFunction(scrollViewRef.current.scrollTo)) {
+            // ListView
+            scrollViewRef.current.scrollTo({x: 0, y: 0, animated: false});
+          } else if (_.isFunction(scrollViewRef.current.scrollToIndex)) {
+            // FlatList
+            scrollViewRef.current.scrollToIndex({index: 0, animated: false});
+          }
+        }
+      }
+    }, [contentSize, layoutSize]);
+
+    const onContentSizeChange = useCallback((contentWidth: number, contentHeight: number) => {
+      const size = props.horizontal ? contentWidth : contentHeight;
+      if (size !== contentSize) {
+        setContentSize(size);
+      }
+      _.invoke(props, 'onContentSizeChange', contentWidth, contentHeight);
+    },
+    [props]);
+
+    const onLayout = useCallback((event) => {
+      const {
+        nativeEvent: {
+          layout: {width, height}
+        }
+      } = event;
+      const size = props.horizontal ? width : height;
+      if (size !== layoutSize) {
+        setLayoutSize(size);
+      }
+      _.invoke(props, 'onLayout', event);
+    },
+    [props]);
+
+    return (
+      <WrappedComponent
+        {...props}
+        onContentSizeChange={onContentSizeChange}
```

Comment:
There might be a little risk with passing the Scroll props as is. with the same names. 
They can be passed in a spread by mistake to places you don't want. also I think we might want to have a more explicit name for them so the user should understand where they are coming from . 
Another option is to group them as an object that the user can spread intentionally 

for instance 
```
<WrappedComponent scrollEnablerProps={{onLayout, scrollEnabled, onContentSizeChange}}
```

Reply:
Done, I've created two use classes, 1 for FlatList and 1 for ListView, they are very similar but I'm not sure how easy they are to combine.